salvador/ContentView.swift

```diff
@@ -9,55 +9,180 @@ import SwiftUI
 
 struct ContentView: View {
     
-    @State var userInputValue = ""
+    @State private var userInputValue = ""
     
     @State var userIsTyping = false
     
- //https://api.openai.com/v1/images/generations
+    @State var image: UIImage? = nil
+    
+    @State var isLoading: Bool  = false
+    
+    //https://api.openai.com/v1/images/generations
     
     var body: some View {
-        ZStack(){
-            VStack {
-                Text("Create an image")
-                    .fontWeight(.black)
-                    .font(.largeTitle)
-                    .padding([.top])
-                
-//                Form{
-                    
-                TextField("Search", text: $userInputValue, onEditingChanged:{
-                    typing in
-                    userIsTyping = typing
-                    
-                })
-                        .textFieldStyle(.roundedBorder)
-                        .padding()
-                        .foregroundColor(.black)
-                        .background(.black.opacity(0.5))
-                
-                        .onSubmit(of: .text){
-                            print(userInputValue)
+        
+        ZStack {
+            NavigationView{
+                Form {
+                    Section{
+                        HStack {
+                            TextField("Generate a image", text: $userInputValue, axis: .vertical)
+                            
+                            
+                            Button("Create"){
+                                isLoading = true
+                                
+                                Task{
+                                    do {
+                                        
+                                        let response = try await createImage.shared.generateImage(withPrompt: userInputValue, apiKey: apiSecret)
+                                        
+                                        
+                                        if let  url = response.data.map(\.url).first {
+                                            
+                                            let (data, _) = try await URLSession.shared.data(from: url)
+                                            
+                                            image  = UIImage(data: data)
+                                            
+                                            isLoading = false
+                                            
+                                            userInputValue = ""
+              
+                                        }
+                                    } catch{
+                                        isLoading = false
+                                        print(error)
+                                    }
+                                }
```

Comment:
This whole button action could be separated from the UI code using MVVM. You could move this code to a view model. Inside the view model you could also move the networking part to a service. The Presentation Layer shouldn't necessarily know details about networking. That will make your code more legible and it will be easier to test it later. Especially when it starts to grow.